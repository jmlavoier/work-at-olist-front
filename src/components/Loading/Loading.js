import { Component, prepareComponent } from 'helpers/engine';
import styles from './Loading.sass';

class Loading extends Component {
  render() {
    this.template`
      <div class="${styles.loading}">
        <span class="${styles.spin1}"></span>
        <span class="${styles.spin2}"></span>
        <span class="${styles.spin3}"></span>
      </div>
    `;
  }
}

export default prepareComponent(Loading);
