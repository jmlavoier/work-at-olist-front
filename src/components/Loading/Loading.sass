@import 'styles/variables'

.loading
  width: 50px
  height: 10px
  display: inline-block

@keyframes spin1-animation
    0%
      height: 10px
      width: 10px
    50%
      height: 4px
      width: 4px
    100%
      height: 10px
      width: 10px

.spin
  border-width: 0
  border-radius: 50%
  display: inline-block
  background-color: $color-default-white
  vertical-align: middle
  animation-duration: 1s
  animation-iteration-count: infinite
  animation-timing-function: ease-in
  animation-name: spin1-animation
  height: 10px
  width: 10px

.spin1
  @extend .spin
  animation-delay: .1s

.spin2
  @extend .spin
  margin-left: 7px
  margin-right: 7px
  animation-delay: .3s

.spin3
  @extend .spin
  animation-delay: .5s
