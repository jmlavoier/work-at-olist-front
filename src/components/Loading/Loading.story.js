import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Loading from './Loading';

storiesOf('Loading', module)
  .add('Default', withInfo()(() => <div style={{
    backgroundColor: '#17d499',
    display: 'inline-block',
    padding: '5px',
  }}><Loading /></div>));
