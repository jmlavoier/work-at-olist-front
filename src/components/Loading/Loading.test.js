import Loading from './Loading';

const ComponentLoading = Loading('loading', {}).component;

const getDOM = (Component) => {
  Component.renderChildrenComponents();
  const html = document.createElement('div');
  html.appendChild(Component.element);
  return {
    el: html.firstElementChild,
    string: html.innerHTML,
  };
}

describe('<Loading />', () => {
  it('Should component match snapshot', () => {
    const component = new ComponentLoading({});
    const wrapper = getDOM(component);

    expect(wrapper.string).toMatchSnapshot();
  });
});
