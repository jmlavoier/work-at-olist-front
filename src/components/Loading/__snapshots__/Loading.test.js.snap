// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> Should component match snapshot 1`] = `
<div
  className="loading"
>
  <span
    className="spin1"
  />
  <span
    className="spin2"
  />
  <span
    className="spin3"
  />
</div>
`;
